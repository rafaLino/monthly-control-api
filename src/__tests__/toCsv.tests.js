import { toCsv } from "../utils/index.mjs";

describe("toCsv tests", () => {
  test("should generate a csv file", () => {
    const data = [
      {
        date: "2024-08",
        id: "123",
        records: {
          incomes: [{ id: "1", name: "bonus", value: "100" }],
          expenses: [{ id: "1", name: "credit card", value: "400" }],
          investments: [{ id: "1", name: "stock", value: "222" }],
        },
      },
      {
        date: "2024-09",
        id: "245",
        records: {
          incomes: [],
          expenses: [],
          investments: [],
        },
      },
    ];
    const result = toCsv(data);

    expect(result).not.toBeNull();
    expect(result).toMatchSnapshot();
  });
});
