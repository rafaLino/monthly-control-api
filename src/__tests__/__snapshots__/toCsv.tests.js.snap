// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toCsv tests should generate a csv file 1`] = `
"date,id,records
2024-08,123,"{""incomes"":[{""id"":""1"",""name"":""bonus"",""value"":""100""}],""expenses"":[{""id"":""1"",""name"":""credit card"",""value"":""400""}],""investments"":[{""id"":""1"",""name"":""stock"",""value"":""222""}]}"
2024-09,245,"{""incomes"":[],""expenses"":[],""investments"":[]}""
`;
